x/gamm/pool-models/balancer/balancer_pool.go

```diff
@@ -380,52 +380,62 @@ func (pa *Pool) updateAllWeights(newWeights []PoolAsset) {
 // PokeTokenWeights checks to see if the pool's token weights need to be updated,
 // and if so, does so.
 func (pa *Pool) PokeTokenWeights(blockTime time.Time) {
-	// Pool weights aren't changing, do nothing.
-	poolWeightsChanging := (pa.PoolParams.SmoothWeightChangeParams != nil)
+	// check if pool weights didn't change
+	poolWeightsChanging := pa.PoolParams.SmoothWeightChangeParams != nil
 	if !poolWeightsChanging {
 		return
 	}
-	// Pool weights are changing.
+
 	// TODO: Add intra-block cache check that we haven't already poked
 	// the block yet.
 	params := *pa.PoolParams.SmoothWeightChangeParams
 
-	// the weights w(t) for the pool at time `t` is the following:
-	//   t <= start_time: w(t) = initial_pool_weights
-	//   start_time < t <= start_time + duration:
+	// The weights w(t) for the pool at time `t` is defined in one of three
+	// possible ways:
+	//
+	// 1. t <= start_time: w(t) = initial_pool_weights
+	//
+	// 2. start_time < t <= start_time + duration:
 	//     w(t) = initial_pool_weights + (t - start_time) *
 	//       (target_pool_weights - initial_pool_weights) / (duration)
-	//   t > start_time + duration: w(t) = target_pool_weights
-
-	// t <= StartTime
-	if blockTime.Before(params.StartTime) || params.StartTime.Equal(blockTime) {
-		// Do nothing
+	//
+	// 3. t > start_time + duration: w(t) = target_pool_weights
+	switch {
+	case blockTime.Before(params.StartTime) || params.StartTime.Equal(blockTime):
```

Comment:
reviewer: no logical changes, just switched the existing branching code to be `switch`-based instead of if/elses.

Reply:
should we generally aim to prefer the code base to be switch based instead of if/elses?